modules/siddhi-extensions/markov-models/src/test/java/org/wso2/siddhi/extension/markov/test/util/SiddhiTestHelper.java

```diff
@@ -0,0 +1,61 @@
+/*
+ *
+ *  * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *  *
+ *  * WSO2 Inc. licenses this file to you under the Apache License,
+ *  * Version 2.0 (the "License"); you may not use this file except
+ *  * in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  *     http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing,
+ *  * software distributed under the License is distributed on an
+ *  * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  * KIND, either express or implied. See the License for the
+ *  * specific language governing permissions and limitations
+ *  * under the License.
+ *
+ */
+
+package org.wso2.siddhi.extension.markov.test.util;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class SiddhiTestHelper {
```

Comment:
@lasanthafdo , I am ok with current implementation, as this is a Test.  CountDownLatch helps to do similar things efficiently without the Thread.sleep(). Its just that I consider Thread.sleep() as bad programming practice. In short, I am OK with current implementation.